c7n/__init__.py

```diff
@@ -1,3 +1,14 @@
 # Copyright The Cloud Custodian Authors.
 # SPDX-License-Identifier: Apache-2.0
 #
+
+import jmespath
+from functools import partial
+from c7n.utils import C7NJmespathFunctions
+
+jmespath.search = partial(
```

Comment:
generally global monkey patching is pretty suspect as it can cause all kinds of unexpected issues.

the alternative would be to have explicit utility function in utils and update relevant call sites to use.

I'll note this is also incompatible with compiled usage, as is common in value filter for example.

adding jmes_search and jmes_compile

Reply:
yea, you're right. updated to use `jmespath_search` and `jmespath_compile` throughout the callsites